source/extensions/common/dynamic_forward_proxy/dns_cache_impl.cc

```diff
@@ -177,7 +180,13 @@ void DnsCacheImpl::startCacheLoad(const std::string& host, uint16_t default_port
   }
 
   primary_host = createHost(host, default_port);
-  startResolve(host, *primary_host);
+  // If the DNS request was simply to create a host endpoint in a Dynamic Forward Proxy cluster,
+  // fast fail the look-up as the address is not needed.
+  if (is_proxy_lookup) {
+    finishResolve(host, Network::DnsResolver::ResolutionStatus::Failure, {}, {}, true);
```

Comment:
This is going to increment the failure stat, right? I think that might be confusing? Should this either not increment that stat or be some kind of success type?